src/plugins/dashboard/public/dashboard_app/dashboard_app.tsx

```diff
@@ -115,11 +115,11 @@ export function DashboardApp({
    * Create options to pass into the dashboard renderer
    */
   const stateFromLocator = loadDashboardHistoryLocationState(getScopedHistory);
-  const getCreationOptions = useCallback((): DashboardCreationOptions => {
+  const getCreationOptions = useCallback((): Promise<DashboardCreationOptions> => {
```

Comment:
I've forgotten why we changed this to be async, in the future we'll want to add panels asynchronously, but is there a reason for the change in this PR? If not, it doesn't need to be undone - just curious!

Reply:
Yup, it was specifically for the `dashboard_with_controls` example - as part of `getCreationOptions`, we are using the  `controlGroupInputBuilder` to add a few controls, and this is done asynchronously. So we needed to make `getCreationOptions` async to allow for this :+1:

Reply:
Ahh yes, of course! Thanks for jogging my memory.